File: csrcs/greeter.cc
#include "greeter.h"
#include <sstream>

std::string Greeter::SimpleGreet() { return "Hello world!"; }

std::map<std::string, std::string>
Greeter::ComplexGreet(const std::vector<std::string> &names,
                      const std::string &common_message) {
  std::map<std::string, std::string> ret;
  for (auto &name : names) {
    std::stringstream ss;
    ss << "Hello, " << name << "! " << common_message;
    ret[name] = ss.str();
  }
  return ret;
}

File: python_template/hello_world/greeter.py
class Greeter:
    def __init__(self) -> None: ...

    def simple_greet(self) -> str:
        return "Hello world!"

    def complex_greet(self, names: list[str], common_message: str) -> dict[str, str]:
        res = {}
        for name in names:
            res[name] = f"Hello, {name}! {common_message}"
        return res
